### PR TITLE
dev: Fix local `just policy-test`

### PR DESCRIPTION
**Pull request by olix0r**
Running `just policy-test` currently fails locally, because it tries to pull in an incorrect cni-plugin image. There are two problems here:

1. The cni-plugin image is published from another repo, so it's not a dependency that should honor the `DOCKER_REGISTRY` setting (which we override when running tests with locally produced images).
2. cni-plugin isn't needed by any of the policy tests.

This change fixes these issues by using the proper image name and splitting out separate, decoupled recipes for fetching the CNI image.

Furethermore, this change updates the treatment of other external
dependencies--specifically proxy-init and prometheus.

### CODE REVIEW

File: justfile [hunk 1]
```diff
@@ -331,8 +328,11 @@ linkerd-load: _linkerd-images _k3d-init
         '{{ controller-image }}:{{ linkerd-tag }}' \
         '{{ policy-controller-image }}:{{ linkerd-tag }}' \
         '{{ proxy-image }}:{{ linkerd-tag }}' \
-        "{{ cni-plugin-image }}:$(yq .image.version charts/linkerd2-cni/values.yaml)" \
-        "{{ proxy-init-image }}:$(yq .proxyInit.image.version charts/linkerd-control-plane/values.yaml)" && exit ; sleep 1 ; done
+        $({{ _proxy-init-image-cmd }}) && exit ; sleep 1 ; done
+
+linkerd-load-cni:
```
Comment (stevej) [inline justfile:333]: This is a definite improvement but what is supposed to call this rule?

**Review by stevej — CHANGES_REQUESTED** [review]

File: justfile (hunk 1 above)
Comment (olix0r) [reply justfile:333]: When does it need to be called? I left it mostly as a convenience for local testing but none of the tests this is used by require CNI.

**Review by hawkw — APPROVED** [review]

**Review by mateiidavid — APPROVED** [review]
